Assets/Scripts/HUD/ProgressBar.cs

```diff
@@ -35,24 +40,64 @@ private void SliderInit()
     {
         slider.maxValue = 100f;
         slider.minValue = 0f;
-        slider.value = 5f;
+        sliderThreshhold = 40f;
+        slider.value = sliderThreshhold;
         fill.color = gradient.Evaluate(0.1f);
+        _SliderText.text = slider.value.ToString("0.00") + "%";
     }
 
     private void Update()
     {
-        //This should be deleted. Only for testing purposes.
-        if (Input.GetKeyDown(KeyCode.Z))
-            ChangeSustainibility(5f);
-        if (Input.GetKeyDown(KeyCode.X))
-            ChangeSustainibility(-5f);
-       
+        decreaseSustainibilityPerSecond(-0.0005f);
```

Comment:
I am not 100% sure how couroutines work, but would it not be better to call this within a couroutine? Because it may be over kill in an Update method if you will only call it every few seconds

Reply:
That causes a lot of bugs :( because decrease sustainibility runs in updates. Unfortunately, that wont work with this functionality.